### ISSUE TITLE
Credentials on scripts imported by importScripts()

### ISSUE THREAD

**Issue by makotoshimazu** [w3c/ServiceWorker#1497]
This is similar to https://github.com/whatwg/html/issues/2557.
Chrome now has a bug around the credentials mode ([issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1042159)), and I'm now wondering what is the correct value for the request's credentials mode.

As far as I read the spec, importScripts() uses [fetch a classic worker-imported script](https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-classic-worker-imported-script) which doesn't set any value to the credentials mode, while [the main script is set to "same-origin"](https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-classic-worker-script). [The default value of the request's credentials mode is "omit"](https://fetch.spec.whatwg.org/#concept-request-credentials-mode), so I think it's "omit".

I implemented the byte-for-byte checking in that way recently, but found that the existing implementation seems to use "include".
What would be the best value for that?

So far, I'm feeling that using "same-origin" would be less confusing.

**Comment by mattto** [comment]
It looks like we changed `fetch()` to use "same-origin" credentials by default (https://github.com/whatwg/fetch/pull/585), probably "importScripts()" in workers should also do that? 

It'd be useful to see what other browsers do here. cc/ @jakearchibald @annevk

**Comment by annevk** [comment]
I would expect `importScripts()` to use "`include`", similar to classic scripts. Or is `importScripts()` different from classic scripts in service workers?

**Comment by annevk** [comment]
cc @domenic

**Comment by domenic** [comment]
I agree with @annevk that I would expect it to use `"include"`. I suspect this was lost in a refactoring. Some research:

- [The earliest HTML commit snapshot](https://html.spec.whatwg.org/commit-snapshots/c9e804f04d03a0658bfa689cb0f368a4d2e37936/#dom-workerglobalscope-importscripts), before integration with the Fetch Standard, uses a "fetch" algorithm defined in HTML. That algorithm sends cookies by default unless invoked with the "block cookies" flag, which is not the case here.
- [This old revision](https://html.spec.whatwg.org/commit-snapshots/18b243aafe328f5cc2095ddfdcdabd585f74b026/#dom-workerglobalscope-importscripts) explicitly sets credentials mode to "include".

I would guess this got lost when introducing module scripts, but I didn't spend the time to confirm.

"same-origin" would be a nice tightening, but probably we should just go back to what the old spec says.

A spec pull request and web platform tests would be very welcome, if you have the time.

**Comment by yoshisatoyanagisawa** [comment]
WPT exists but the spec looks not updated yet?
https://wpt.fyi/results/service-workers/service-worker/credentials.https.html

**Comment by monica-ch** [comment]
[Bug triage] Needs spec update by adding "credentials-mode" to "include"
